examples/step-89/doc/intro.dox

```diff
@@ -0,0 +1,215 @@
+enforces the coupling via Lagrange multipliers. Obtained intersections are also referred to as
+mortars. On each mortar a new integration rule is defined. The integral of the face of element
+$K^-$ is computed on the intersections. This way, the numerical integration is exact as long
+as a sufficient number of integration points is used.
+In this tutorial, the intersections are computed using `CGAL`. Therefore, `deal.II` has
+to be configured with `DEAL_II_WITH_CGAL` for the Nitsche-type mortaring implementation.
+
+<h3>%FERemoteEvaluation</h3>
+
+FERemoteEvaluation is a wrapper class which provides a similar interface to, e.g., FEEvaluation to access values over non-matching interfaces in matrix-free loops.
+A detailed description on how to setup the class and how to use it in actual code is given below on hands-on examples.
+Within this tutorial we only show the usage for non-matching discretizations.
+Note however, that FERemoteEvaluation can also be used in other settings such as volume coupling.
+Under the hood, Utilities::MPI::RemotePointEvaluation is used to query the solution or gradients in quadrature points.
+A detailled description how this is done can be found in step-87.
+The main difference in the usage of FERemoteEvaluation compared to FEEvaluation is that values/gradients are cached <i>before</i> each loop and only accessed inside the loop.
+I.e., the code differs as follows: On the one hand, the standard code to evaluate fluxes via FEEvaluation
+@code
+const auto face_function =
+  [&](const auto &data, auto &dst, const auto &src, const auto face_range) {
+
+    FEFaceEvaluation phi_m(data, true);
+    FEFaceEvaluation phi_p(data, false);
+
+    for (unsigned int f = face_range.first; f < face_range.second; ++f)
+    {
+      phi_m.reinit(f);
+      phi_p.reinit(f);
+
+      phi_p.gather_evaluate(src, EvaluationFlags::values); //compute values on face f
+
+      for (unsigned int q = 0; q < phi_m.n_q_points; ++q)
+        phi_m.submit_value(phi_p.get_value(q), q); //access values with phi_p
+
+      phi_m.integrate_scatter(EvaluationFlags::values, dst);
+     }
+   };
+
+matrix_free.template loop<VectorType, VectorType>({}, face_function, {}, dst, src);
+@endcode
+On the other hand, the code to evaluate fluxes over non-matching faces. Lets assume all boundary faces are somehow connected non-conformingly to be able to focus
+on the relevant differences.
+
+@code
+// Initialize FERemoteEvaluation: Note, that FERemoteEvaluation internally manages
+// the memory to store cached values. Therefore, FERemoteEvaluation
+// should be initialized only once to avoid continuous memory
+// allocation/deallocation. At this point, remote_communicator is assumed
+// to be initialized.
+FERemoteEvaluation<dim,Number> phi_p(remote_communicator);
+
+// Cache needed values before the actual matrix-free loop (on one thread).
+phi_p_evaluator.gather_evaluate(src, EvaluationFlags::values);
```

Comment:
Is the information regarding "one thread" important? I think the important information is that the information is computed globally and by collecting the information via the necessary communication and all other magic. I would personally not call the process "cache" but simply "Precompute the interpolated values of the finite element solution at all the quadrature points outside the loop, invoking the vector entries and respective basis function at possibly remote MPI processes before communication."

Reply:
This is much better :+1: